plugins/linuxplugin/interface_config.go

```diff
@@ -503,24 +514,24 @@ func (plugin *LinuxInterfaceConfigurator) removeObsoleteVeth(nsMgmtCtx *linuxcal
 
 // addVethInterface creates a new VETH interface with a "clean" configuration.
 func (plugin *LinuxInterfaceConfigurator) addVethInterface(nsMgmtCtx *linuxcalls.NamespaceMgmtCtx, iface *intf.LinuxInterfaces_Interface, peer *intf.LinuxInterfaces_Interface) error {
-	err := plugin.removeObsoleteVeth(nsMgmtCtx, iface.Name, iface.Namespace)
+	err := plugin.removeObsoleteVeth(nsMgmtCtx, iface.HostIfName, iface.Namespace)
```

Comment:
removeObsoleteVeth() needs both the logical name (for calls to plugin.ifIndexes) and the physical name (for linuxcalls).

Also please remove:
plugin.ifIndexes.UnregisterName(peerName)
from removeObsoleteVeth(). It requires the peer logical name and not the physical one obtained with linuxcalls.GetVethPeerName(). There are always successive calls to removeObsoleteVeth() for both ends anyway.